go/vt/vttablet/common/flags.go

```diff
@@ -94,4 +95,6 @@ func registerFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&vreplicationStoreCompressedGTID, "vreplication_store_compressed_gtid", vreplicationStoreCompressedGTID, "Store compressed gtids in the pos column of the sidecar database's vreplication table")
 
 	fs.IntVar(&vreplicationParallelInsertWorkers, "vreplication-parallel-insert-workers", vreplicationParallelInsertWorkers, "Number of parallel insertion workers to use during copy phase. Set <= 1 to disable parallelism, or > 1 to enable concurrent insertion during copy phase.")
+
+	fs.Uint64Var(&mysql.ZstdInMemoryDecompressorMaxSize, "binlog-in-memory-decompressor-max-size", mysql.ZstdInMemoryDecompressorMaxSize, "This value sets the uncompressed transaction payload size at which we switch from in-memory buffer based decompression to the slower streaming mode.")
```

Comment:
Let's add validation to the entered value - there have to be some bounds I guess? And first read it into a local variable before assigning it into `mysql.ZstdInMemoryDecompressorMaxSize`.

Reply:
The bounds are set by the type. If it’s 0 then streaming mode will always be used. If it’s the max then the in-memory buffers will always be used.

Why would we read it into a local variable first?